src/contract_wrappers/tokens/security_token_wrapper.ts

```diff
@@ -903,17 +1254,44 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
     );
   };
 
-  public burnWithData = async (params: BurnWithDataParams) => {
+  public redeem = async (params: RedeemParams) => {
     await this.checkBalanceFromGreaterThanValue((await this.web3Wrapper.getAvailableAddressesAsync())[0], params.value);
-    return (await this.contract).burnWithData.sendTransactionAsync(
+    return (await this.contract).redeem.sendTransactionAsync(
       valueToWei(params.value, await this.decimals()),
       stringToBytes32(params.data),
       params.txData,
       params.safetyFactor,
     );
   };
 
-  public burnFromWithData = async (params: BurnFromWithDataParams) => {
+  public redeemByPartition = async (params: RedeemByPartitionParams) => {
+    await this.checkBalanceFromGreaterThanValue((await this.web3Wrapper.getAvailableAddressesAsync())[0], params.value);
+    assert.assert(params.partition === 'UNLOCKED', 'Invalid Partition');
+    return (await this.contract).redeemByPartition.sendTransactionAsync(
+      params.partition,
+      valueToWei(params.value, await this.decimals()),
+      stringToBytes32(params.data),
+      params.txData,
+      params.safetyFactor,
+    );
+  };
+
+  public operatorRedeemByPartition = async (params: OperatorRedeemByPartitionParams) => {
+    await this.checkBalanceFromGreaterThanValue((await this.web3Wrapper.getAvailableAddressesAsync())[0], params.value);
+    assert.isNonZeroETHAddressHex('TokenHolder', params.tokenHolder);
+    assert.assert(params.partition === 'UNLOCKED', 'Invalid Partition');
```

Comment:
Same as above, partitions should be part of an enum

Also, this check should probably be done on a separate function `isValidPartition` that compares the value to all possible partition values. This will make this easier to extend if/when more partitions are added